### PR TITLE
Typescript toolchain fixes and improvements

### PR DESCRIPTION
**Pull request by toninorair**
- Add unsupported types of declarations;
- Support for docstrings;
- Multiple fixes after project testing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,84 @@
...
-interface Foo1 {
-    bar: {
-        (s: string): string;
-        //(n: number): string;
-    }
-}
-var foo_inst: Foo1 = { bar(s: string): string { return "1"; } }
-```
+**Successfully:**
+* [n3-charts/line-chart](https://github.com/n3-charts/line-chart)
```
Comment (alexsaveliev) [inline README.md]: Please use sourcegraph.com/foo/bar instead of github.com/foo/bar

File: README.md [hunk 2]
```diff
@@ -1,34 +1,84 @@
...
+**srclib-typescript** is a [srclib](https://sourcegraph.com/sourcegraph/srclib)
+toolchain that performs TypeScript code analysis: type inference,
+documentation generation, jump-to-definition, dependency resolution, etc.
+It enables this functionality in any client application whose code analysis is
+powered by srclib, including [Sourcegraph](https://sourcegraph.com).
+## Installation
+This toolchain is not a standalone program; it provides additional functionality
+to applications that use [srclib](https://srclib.org).
+First,
+[install the `src` program (see srclib installation instructions)](https://sourcegraph.com/sourcegraph/srclib).
```
Comment (alexsaveliev) [inline README.md]: I think there should be `srclib`

File: README.md [hunk 3]
```diff
@@ -1,34 +1,84 @@
...
+Then run:
+```
+git clone https://github.com/sourcegraph/srclib-typescript.git
+cd srclib-typescript
+src toolchain add sourcegraph.com/sourcegraph/srclib-typescript
```
Comment (alexsaveliev) [inline README.md]: There is no `srclib toolchain add` anymore, please propose a pull request that will support `srclib toolchain install typescript`

File: README.md [hunk 4]
```diff
@@ -1,34 +1,84 @@
...
+```
+To verify that installation succeeded, run:
+```
+src toolchain list
```
Comment (alexsaveliev) [inline README.md]: Please replace `src` with `srclib`

File: generate-tsconfig.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash 
+# This script generates tsconfig.json file, it looks for all files with extension .ts and adds them to the file section. 
+# Old tsconfig.json file, if exists is saved with .ol.timestamp extension
```
Comment (alexsaveliev) [inline generate-tsconfig.sh:5]: Please mention that this script is for testing purposes only and has no impact on typescript toolchain

File: README.md [hunk 6]
```diff
@@ -1,34 +1,84 @@
...
+* [DefinitelyTyped/DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) 
 
+**Next issues were found:**
+* Unsupported builtins for interfaces.
+* Partially supported shorthand property assignments.
+* ExportSpecifiers can work better.
+* Import clause is not fully supported (syntax tree kind = 226, reproduced on [doppio](https://github.com/plasma-umass/doppio) project).
 
+## Run procedure
+Testing this toolchain requires that you have installed `src` from
```
Comment (alexsaveliev) [inline README.md]: Please replace `src` with `srclib`

File: README.md [hunk 7]
```diff
@@ -1,34 +1,84 @@
...
+[srclib](https://sourcegraph.com/sourcegraph/srclib) and that you have this
+toolchain set up. See srclib documentation for more information.
+To test this toolchain's output against the expected output, run:
```
Comment (alexsaveliev) [inline README.md:70]: `srclib` (former `src`) does not support `-m` flag anymore. Also, there is no need to build Docker image before running the tests

File: src/utils.ts [hunk 8]
```diff
@@ -6,7 +6,7 @@ export class DefKind {
     static FUNC: string = "function";
     static METHOD: string = "method";
     static VAR: string = "var";
-    static PARAM: string = "param";
+    static PARAM: string = "paramater";
```
Comment (alexsaveliev) [inline src/utils.ts]: Please fix the typo

**Comment by alexsaveliev** [comment]
Please take into account my remarks. Also, could you please document the code (classes and methods)? 

Thanks

**Comment by beyang** [comment]
Can this be merged?

**Pull request merged**